core/lib/spree/permission_sets/default_customer.rb

```diff
@@ -11,6 +11,9 @@ def activate!
         can [:read, :update], Order do |order, token|
           order.user == user || (order.guest_token.present? && token == order.guest_token)
         end
+        cannot :update, Order do |order|
+          order.completed?
```

Comment:
Could this be done as `cannot :update, Order, &:completed?`

Reply:
Sure, but I prefer this way for clarity and cleaner git history if the block content changes in the future.